### PR TITLE
Implement basic ServiceNow incident creation

### PR DESCRIPTION
**Pull request by louisfelix**
Implement basic ServiceNow incident creation.
Also use Prometheus logging libs.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -49,17 +49,17 @@ func webhook(w http.ResponseWriter, r *http.Request) {
...
 	kingpin.HelpFlag.Short('h')
 	kingpin.Parse()
 
-	log.Println("Starting webhook", version.Info())
-	log.Println("Build context", version.BuildContext())
+	log.Info("Starting webhook", version.Info())
+	log.Info("Build context", version.BuildContext())
 
 	http.HandleFunc("/webhook", webhook)
 	http.Handle("/metrics", promhttp.Handler())
```
Comment (vincentcreusotfx) [inline main.go]: Can you tell me why we have this ? /metrics is served for what actually ?

File: servicenow.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		baseURL:    fmt.Sprintf(serviceNowBaseURL, instanceName),
+		authHeader: fmt.Sprintf("Basic %s", base64.URLEncoding.EncodeToString([]byte(userName+":"+password))),
+		client:     http.DefaultClient,
+	}, nil
+}
+
+// Create a table item in ServiceNow from a post body
+func (snClient *ServiceNowClient) create(table string, body []byte) (string, error) {
+	log.Infof("Creating a ServiceNow %s", table)
+	url := fmt.Sprintf("%s/api/now/table/%s", snClient.baseURL, table)
```
Comment (vincentcreusotfx) [inline servicenow.go]: could you externalize url strings ? constantize it

File: servicenow.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		log.Errorf("Error creating the request. %s", err)
+		return "", err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", snClient.authHeader)
+	resp, err := snClient.client.Do(req)
+	if err != nil {
```
Comment (vincentcreusotfx) [inline servicenow.go]: Could you add tests to go to that error point for example ?

File: main.go (hunk 1 above)
Comment (thomaslacher) [reply main.go]: I think it's used to fetch metrics from the exporter directly !

File: servicenow.go (hunk 2 above)
Comment (thomaslacher) [reply servicenow.go]: I think we should use the constant tableAPI here

File: main.go (hunk 1 above)
Comment (louisfelix) [reply main.go]: Exactly. This is for the monitoring of the alertanager webhook.

File: servicenow.go (hunk 2 above)
Comment (louisfelix) [reply servicenow.go]: I will take a look, yes!

File: servicenow.go (hunk 3 above)
Comment (louisfelix) [reply servicenow.go]: I will take a look, yes!

File: main.go (hunk 1 above)
Comment (vincentcreusotfx) [reply main.go]: Ok I get it thanks

File: servicenow.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/prometheus/common/log"
+)
+
+const (
+	serviceNowBaseURL = "https://%s.service-now.com"
+	tableAPI          = "%s/api/now/table/%s"
```
Comment (thomaslacher) [inline servicenow.go]: Should we specify the API version here ? 
With that string it will always call latest !

Comment (louisfelix) [reply servicenow.go]: Good point, I'll add it.

File: servicenow.go (hunk 2 above)
Comment (louisfelix) [reply servicenow.go]: Fixed

File: servicenow.go (hunk 4 above)
Comment (louisfelix) [reply servicenow.go]: Fixed.

File: servicenow.go (hunk 3 above)
Comment (louisfelix) [reply servicenow.go]: I will try to add more tests in next PR ;-)

**Pull request merged**